New administration chapter:

cwickert opened the pull request:
This chapter contains 4 sections:

* Registration (general explanation)
* Registration with registercloudguest
* Registration with SUSEConnect
* Deregistering instances

cwickert committed:
New section on registercloudguest (WIP)

cwickert committed:
Administration: Add registration with SUSEConnect

(sec-yast-software-register-suseconnect from the Deployment Guide)

cwickert committed:
Administraion: New section 'Deregistering instances'

cwickert committed:
Administration: Polish 'registercloudguest' section

cwickert committed:
Administration: spellcheck

cwickert committed:
Registering: Do not use gerund for task topics

cwickert committed:
Register with SUSEConnect: Polishing

cwickert commented:
@kvanderveer, here you are a PDF version: [cha-admin_color_en.pdf](https://github.com/SUSE/doc-public-cloud/files/8072324/cha-admin_color_en.pdf)

cwickert committed:
Register with registercloudguest: Make command and wording consitent with SUSEConnect instructions

cwickert committed:
Register with registercloudguest: two wording fixes for procedure

cwickert committed:
Register with registercloudguest: Remove hardcoded sudo

kvanderveer commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,184 @@
+   termination with:
+  </para>
+  <screen><command>registercloudguest</command> <option>--clean</option></screen>
+  <para>
```
We only recommend they do this if they have registered using registercloudguest, correct @rjschwei ? I imagine if they registered with SUSEConnect, they still should use SUSEConnect -d. Are there commands they can run to see the status of whether they are registered as expected or not? I installed registercloudguest but didn't see anything.

cwickert replied:
Good point. We certainly need to mention SUSEConnect here, too. But that means we also need to document the best way to determine _if_ a system is registered correctly and _how_ it was registered (registercloudguest or SUSEConnect)

rjschwei replied:
When a system is decommissioned it should always be removed from SCC to keep the count for th ecustomer accurate. This happens either with `registercloudguest --clean` or `SUSEConnect -d`. The latter only works if cloud-regionsrv-client is not installed. SO the general recommendation here is correct, but we probably want more details.

cwickert replied:
In order to get this published, I suggest we stick to what we have. It's correct, it covers both `registercloudguest` and `SUSEConnect`, it just lacks some detail that we can still add later. If you consider this a blocker, please let me know what you want me to add.

cwickert replied:
I've updated the deregistration instructions in b9c34feeffa88ba0cd2235ee962c45d78fb09e8e. Hope they are ok.

kvanderveer commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,184 @@
+   counted against your subscription.
+  </para>
+ </sect1>
+
```
Re: Troubleshooting, what I'm hearing is that they should run registercloudguest --clean and try again. If that fails, they can try the SUSEConnect method (remembering to de-register appropriately, I imagine.)

We could also point them to https://scc.suse.com/docs/help or other resources to see if they are registered properly.

rjschwei replied:
No need for repeat commands.

cwickert replied:
I'm afraid we have to repeat the commands here. Deregistering an instance when it gets decommissioned and reregiering it because of problems are two separate things. Most users these days access our documentation from a search engine, means they search for deregistering or reregistering. We have no control where they land and how much text they actually read, so each section needs to be self-contained.

cwickert committed:
Deregister: mention SUSEConnect, too

cwickert committed:
Consistently use suseconnect entity

cwickert committed:
Make usage of <command> in screens consistent

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+  <para>
+   Like other &suse; products, &sle; in public clouds has to be registered to
+   receive updates. There are different ways to register, depending on what
+   instance you register where.
```
Suggest:

"Registration options are determined by the image flavor (crosslink to the images_chapter) and the primary product of the image."

cwickert replied:
What is the "primary product"? We are not using that term yet and would need to explain it. Are there different registration requirements for different products? Would that mean that e.g. SLES-SAP cannot be registered with registercloudguest as the 'SAP Applications' module requires an additional / a different subscription?

rjschwei replied:
SUMa images have 2 SUSE products, SLES & SUMA Server or Suma Proxy. The primary product in those images is SUMa.

We do, at this point not support to register SUMa against the update infrastructure.

cwickert replied:
Thanks, I've added SUMa to the list of products currently not supported by registercloudguest in 68a7cf571dbd11fad31d9f7190b7aece7c7cce72. However I'd like to give an explanation of why it's not supported (I did the same for SLE Micro and chost images)

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+   <listitem>
+    <para>
+     <literal>BYOS</literal> instances have to be registered manually with your
+     &suse; registration key. You can register against a local update server,
```
s/a local update server/an update server that is part of the SUSE update infrastructure in the framework/

I think it is important that we are more explicit about this. One could easily interpret that "a local update server" is one that the customer has to run. While this point is covered later in the same sentence I think being more explicit is desirable.

cwickert replied:
Applied, thanks.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+  <itemizedlist>
+   <listitem>
+    <para>
+     Images with a datestamp <emphasis>later</emphasis> than
```
/Images/BYOS images/

cwickert replied:
We already have that right above the list: "There are two different ways to register _BYOS_ instances". I don't think we need to repeat it here already as the list has only two entries.

cwickert replied:
Applied it anyway for clarity.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+     <literal>20220103</literal> can be registered with
+     <command>registercloudguest</command>. This sets up
+     <command>&suseconnect;</command> to work against the cloud framework's update
+     infrastructure rather than the &scc;, resulting in significantly faster
```
This is inconsistent with the intro. In the intro we state `There are two different ways to register BYOS instances with the &scc;,` but this says `...rather than the &scc;,`

We have to decide if we consider the registration using the update infrastructure as a proxy to be against scc or not and then word the text accordingly.

I we declare that the use of `registercloudguest` is not a registration against scc, as implied here, then the intro has to change. If we decide the registration to the update infrastructure is against SCC as the registration information gets forwarded to scc then this part has to change. In any event we should state that the registration data is forwarded to SCC.

cwickert replied:
Thanks, this is an important point. My suggestion:

* Registrations against the cloud's update infrastructure are considered against SCC. They show up in SCC, so from a customer's perspective, they are registered there, regardless of their update servers
* Registrations with SUSEConnect against SCC are considered direct registrations, means we would write "registered against SCC _directly_".

Would that work or do you think it still leaves too much room for misunderstandings?

jesusbv replied:
In the first option: should we not add/should it not say that that registration is using registercloudguest ?
Similar to the second option that explicitly mentions the registration is made with SUSEConnect unless the previous sentence has not changed (`can be registered with <command>registercloudguest...`) but maybe be more explicit about what means/does what

cwickert replied:
I've changed my mind here. Instead of direct (against SCC) and indirect registrations (against the
cloud update servers), I tried to make things more straightforward:
* No longer use "register against SCC" be default, just "register".
* The registration 'target' now determines what we are talking
  about:
  * SUSEConnect registers against SCC, RMT or SUMA.
  * registercloudguest registers against the cloud update infrastructure
    and forwards registration information to the SCC.
  * Only use 'against SCC' if we are really talking about SCC.

For details, check the changes in bed45a53d44ac18008b847470b9777872f2186d2

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+   </listitem>
+   <listitem>
+    <para>
+     Images with a datestamp <emphasis>prior</emphasis> to
```
s/Images/BYOS images/

cwickert replied:
Again, we already have that right above the list.

cwickert replied:
Applied nevertheless.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+     <command>&suseconnect;</command> to work against the cloud framework's update
+     infrastructure rather than the &scc;, resulting in significantly faster
+     package downloads. Refer to
+     <xref linkend="sec-admin-registercloudguest"/> for instructions.
```
I think we should also point out that the user can use SUSEConnect directly if they want to register to scc for some reason.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+     <package>&suseconnect;</package>. Refer to <xref
+      linkend="sec-admin-suseconnect"/>. Alternatively, you can install
+     <command>registercloudguest</command> to benefit from faster downloads
+     from the local cloud update infrastructure.
```
Depending on what we want to emphasis I think it would be advantages to turn this around and first speak of the requirement to install cloud-regionsrv-client, could read as follows:

"""
BYOS images with datestamp prior to 20220103 do not have the client side packages installed to register against the SUSE update images. If you want to register against the update infrastructure install

cloud-regionsrv-client, cloud-regionsrv-client-addon-azure,cloud-regionsrv-client-plugin-azure, regionServiceClientConfigAzure for Azure

cloud-regionsrv-client, cloud-regionsrv-client-plugin-ec2, regionServiceClientConfigEC2 for AWS

and

cloud-regionsrv-client, cloud-regionsrv-client-plugin-gce, regionServiceClientConfigGCE for GCE

to register against scc use SUSEConnect.
"""

In any event the install of "<command>registercloudguest</command>" is not sufficient as that would only bring the "cloud-regionsrv-client" package and that will not be enough as shown in the suggested change text.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+  </itemizedlist>
+  <para>
+   In summary, use <command>registercloudguest</command> to register with the
+   &scc;; it provides significant benefits even for older instances.
```
Here is another inconsistency that will need to be fixed depending on our decision about where the registration via the update infrastructure is to SCC or not.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+   In summary, use <command>registercloudguest</command> to register with the
+   &scc;; it provides significant benefits even for older instances.
+   Use <command>&suseconnect;</command> to register with or your own &susemgr;
+   or &rmt; infrastructure.
```
and SUSEConnect can still be used to register to SCC directly.

cwickert replied:
It can, but this summary is our recommendation. If there is no reason to use SUSEConnect and SCC, we should recommend registercloudguest and the cloud update infrastructure for faster downloads. Right?

cwickert replied:
Added in 3bcfc5f165255f7e860343998dc073de8f98c024.

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+  <sect2 xml:id="sec-admin-registercloudguest">
+   <title>Register with <command>registercloudguest</command></title>
+   <para>
+    To register a BYOS instance against the &scc; with
```
Another place of inconsistency of scc vs. the update infrastructure.

To avoid misunderstandings let me clarify what happens:

When the user uses `registercloudguest` the user registers against the update infrastructure. The registration information is forwarded to SCC such that the user can see the system in the scc UI. Instead of using `registercloudguest` for the initial registration the user can use `SUSEConnect` which would then register the system against scc, i.e. updates.suse.com.

What is important is **the initial registration** has to be done with either one tool or the other and that determines what happens with the rest.

cwickert replied:
Ok, but that raises two questions:

1. What is the best way to reliably determine which tool was used for the initial registration?
2. Should we recommend a tool, e.g. should we recommend registercloudguest because of the faster downloads?

rjschwei commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+       <para>For &gce; run:</para>
+<screen>&prompt.root;<command>zypper</command> in cloud-regionsrv-client cloud-regionsrv-client-plugin-gce \
+ regionServiceClientConfigGCE regionServiceCertsGCE</screen>
+      </step>
```
OK this covers what I wrote earlier, so in the previous re-phrasing proposal we could just cross reference to here.

jesusbv commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+     </para>
+<screen>&prompt.root;<command>registercloudguest</command> <option>--clean</option></screen>
+     <para>
+      Do <emphasis>not</emphasis> use <command>&suseconnect; -d</command>; it will
```
Not sure if it is worthy to mention that it will no longer work _if_ registercloudguest is installed, no matter what the system used to get registered (registercloudguest or SUSEConnect)

cwickert replied:
IMHO this is covered. The instructions are step by step, so you already have registercloudguest installed at that point and we say that SUSEConnect will no longer work.

cwickert replied:
I've now included this in b9c34feeffa88ba0cd2235ee962c45d78fb09e8e.

jesusbv commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,187 @@
+  <para>
+   If a system was registered with <command>&suseconnect;</command>, run:
+  </para>
+  <screen>&prompt.root;<command>&suseconnect;</command> <option>-d</option></screen>
```
Same as I mentioned before, maybe being more explicit about the commands, if the system was registered with SUSEConnect but registercloudguest is installed, this will **not** work (and it will show a message suggesting to use registercloudguest --clean but we can avoid the user going that path)

cwickert replied:
@jesusbv, that makes sense. We could then consolidate the troubleshooting in one procedure. But this means, we need

- a simple command to get the current registration status (step one)
- a way to determine what tool was used for the registration (step two)

cwickert replied:
So for the registration status, I used `SUSEConnect --status-text`. To determine what tool was used I check for presence of /var/log/guestregister. If the file is present I assume registercloudguest, if not SUSEConnect.

Is this save? Could we run into a corner case, say a system was registered with SUSEConnect, but cloud-regionsrv-client was installed afterwards, something called registercloudguest and created the log file?

cwickert committed:
1st round of review feedback by @rjshwei

cwickert committed:
Registering instances: Make wording more precise and consistent

Instead of direct (against SCC) and indirect registrations (against the
cloud update server), I tried to make things more straightforward:
* No longer use "register against SCC" be default, just "register".
* Instead, the registration 'target' now determines what we are talking
  about:
  * SUSEConnect registers against SCC, RMT or SUMA.
  * registercloudguest registers against the cloud update infrastructure
    and forwards registration information to the SCC.
  * Only use 'against SCC' if we are really talking about SCC.

cwickert committed:
doc-kit update

cwickert committed:
Remove useless product entity

cwickert committed:
Registration: List products/flavors not supported by registercloudguest

cwickert committed:
Update deregistration instructions and turn them into a procedure

cwickert committed:
Deregistration: Fix package name

cwickert commented:
Hi @rjschwei, I think I addressed all of the feedback. For the changes I applied, I marked our conversations as resolved, but left the ones open where I still have questions. Please re-review.

cwickert committed:
Registration: Mention that SLE Micro does not include registercloudguest

cwickert committed:
Registration: Do not recommend registercloudguest, focus on the different registration 'targets'.

jesusbv commented on xml/cha_administration.xml:
```diff
@@ -28,4 +28,246 @@
+      <screen>&prompt.root;<command>&suseconnect;</command> <option>-d</option></screen>
+      <para>
+       If this does not work, make sure the package
+       <package>cloud-regionsrv-client</package> is not be installed. It may have been
```
s/is not be installed/is not installed

cwickert replied:
Fixed in d5755c19

cwickert committed:
Deregister instances: Remove supoerflous word

cwickert committed:
Register with registercloudguest: Remove another superflous word

cwickert committed:
Register with registercloudguest: Bump SLE version (15 SP3 -> 15 SP4

cwickert commented:
@rjschwei, I think I addressed all your comments and really need to merge this now. Even if the content may be inaccurate or there are some gaps, it's not wrong or misleading any way better than what we have now, because right now we have nothing.

If you have any concerns, we can still address them in a separate issue/PR. Let's improve iteratively instead of creating one big chunk that is supposed to deal with everything.
